spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java

```diff
@@ -0,0 +1,24 @@
+package com.baeldung.sample.singleton;
+
+public final class SingletonPattern {
+
+    private static volatile SingletonPattern instance = null;
+
+    private SingletonPattern() {}
+
+    public static SingletonPattern getInstance() {
+        if (instance == null) {
+            synchronized(SingletonPattern.class) {
+                if (instance == null) {
+                    instance = new SingletonPattern();
```

Comment:
I'd be interested in comparing this to 

```java
public final class SingletonPattern {
   private static final SingletonPattern instance = new SingletonPattern();

   public static SingletonPattern getInstance() { return instance; }
}
```

I believe that these are, functionally, the same!!!

Reply:
Hi @ashleyfrieze, functionally I would say yes. But your example is not using lazy loading, which is described in the article. I would leave it like this.